docs/latest/design/egctl.md

```diff
@@ -0,0 +1,61 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [entity] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `config`, `version`.
+
+* `entity`: Specifies the entity the operation is being performed on such as `envoy-proxy` or `envoy-gateway`.
+
+* `name`: Specifies the name of the specified instance. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-c` or `--config` flags to specify the values for installing.
+
+If you need help, run `egctl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation | Syntax                           | Description                                                                 |
+| --------- | -------------------------------- | --------------------------------------------------------------------------- |
+| `version` | `egctl version`                  | Prints out build version information.                                       |
+| `config`  | `egctl config ENTITY`            | Retrieve information about proxy configuration from envoy proxy and gateway |
```

Comment:
As previously mentioned, `config` should be used to manage the egctl config and not the proxy configuration.

Reply:
the PR shares some examples on how `config` is used
```
# Retrieve all information about proxy configuration from envoy
 egctl config envoy-proxy all <instance_name>

 # Retrieve listener information about proxy configuration from envoy 
 egctl config envoy-proxy listener <instance_name>

 # Retrieve information about envoy gateway
 egctl config envoy-gateway
 ```

Reply:
@arkodg this is where I disagree. The `config` subcommand should be used to configure egctl (similar to `kubectl config`). To manage the proxy config, I prefer `egctl get proxy bootstrap|clusters|routes|etc` but I'm open to `egctl proxy-config bootstrap|clusters|routes|etc`. To support similar functionality for Envoy Gateway in the future, the `egctl gateway-config or control-plane-config` subcommand can be introduced. If you don't agree with this proposal, how do you suggest managing the egctl configuration, i.e. changing between EG endpoints?

Reply:
flags such as `--context` (or `kubeconfig`) can be used to change the server endpoint